build.rs

```diff
@@ -11,20 +11,48 @@ fn main() {
         .file("./deps/ada.cpp")
         .include("./deps/ada.h")
         .include("./deps/ada_c.h")
-        .cpp(true);
+        .cpp(true)
+        .std("c++17");
 
+    let compile_target_arch = env::var("CARGO_CFG_TARGET_ARCH").expect("CARGO_CFG_TARGET_ARCH");
     let compile_target_os = env::var("CARGO_CFG_TARGET_OS").expect("CARGO_CFG_TARGET_OS");
     let compile_target_env = env::var("CARGO_CFG_TARGET_ENV").expect("CARGO_CFG_TARGET_ENV");
-    if !(compile_target_os == "windows" && compile_target_env == "msvc") {
-        build.compiler("clang++");
-        build.flag("-std=c++17");
-        #[cfg(feature = "libcpp")]
-        {
-            build.cpp_set_stdlib("c++");
-            println!("cargo:rustc-link-lib=c++");
+    // Except for Emscripten target (which emulates POSIX environment), compile to Wasm via WASI SDK
+    // which is currently the only standalone provider of stdlib for compilation of C/C++ libraries.
+    if compile_target_arch.starts_with("wasm") && compile_target_os != "emscripten" {
```

Comment:
Can you update github workflow to build wasm too?

Reply:
Sure.